deploy/cmd/create.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/spf13/viper"
+	"golang.org/x/sync/errgroup"
+)
+
+// createCmd represents the create command
+func createCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [sdl-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Create a deployment to be managed by the deploy application",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			log := logger.With("cli", "create")
+			dd, err := NewDeploymentData(args[0], cmd.Flags(), clientCtx)
+			if err != nil {
+				return err
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			group, _ := errgroup.WithContext(ctx)
+
+			// Listen to on chain events and send the manifest when required
+			group.Go(func() error {
+				if err = ChainEmitter(ctx, clientCtx, DeploymentDataUpdateHandler(dd), SendManifestHander(clientCtx, dd)); err != nil {
+					log.Error("error watching events", err)
+				}
+				return err
+			})
+
+			// Send the deployment creation transaction
+			group.Go(func() error {
+				if err = TxCreateDeployment(clientCtx, cmd.Flags(), dd); err != nil {
+					log.Error("error creating deployment", err)
+				}
+				return err
+			})
+
+			// Wait for the leases to be created and then start polling the provider for service availability
+			group.Go(func() error {
+				if err = WaitForLeasesAndPollService(clientCtx, dd, cancel); err != nil {
+					log.Error("error listening for service", err)
+				}
+				return err
+			})
+
+			return group.Wait()
+		},
+	}
+
+	cmd.Flags().String(flags.FlagChainID, "", "The network chain ID")
+	if err := viper.BindPFlag(flags.FlagChainID, cmd.Flags().Lookup(flags.FlagChainID)); err != nil {
+		return nil
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	dcli.AddDeploymentIDFlags(cmd.Flags())
+
+	return cmd
+}
+
+// WaitForLeasesAndPollService waits for leases
+func WaitForLeasesAndPollService(clientCtx client.Context, dd *DeploymentData, cancel context.CancelFunc) error {
+	log := logger
+	pclient := pmodule.AppModuleBasic{}.GetQueryClient(clientCtx)
+	timeout := time.After(150 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		case <-timeout:
+			log.Info("timed out (90s) listening for deployment to be available")
+			cancel()
+			return nil
+		case <-tick:
+			if dd.ExpectedLeases() {
+				for _, l := range dd.Leases() {
+
+					var (
+						p   *ptypes.Provider
+						err error
+					)
+					if err := retry.Do(func() error {
+						res, err := pclient.Provider(
+							context.Background(),
+							&ptypes.QueryProviderRequest{Owner: l.Provider},
+						)
+						if err != nil {
+							// TODO: Log retry?
+							return err
+						}
+
+						p = &res.Provider
+						return nil
+					}); err != nil {
+						cancel()
+						return fmt.Errorf("error querying provider: %w", err)
+					}
+
+					// TODO: Move to using service status here?
```

Comment:
Yes, after the lease status is OK it would make sense to run service status and when that is ready show that result to the user. This should have the information about what host / port they need to access their deployment on.